### PR TITLE
techdebt/ab2d 3291 version enum change for r4

### PR DESCRIPTION
**Pull request by sharshar-sb**
**JIRA Tickets:**

[AB2D-3291](https://jira.cms.gov/browse/AB2D-3291) - Create an enum for FHIR version
 
### What Does This PR Do?

This does a bunch of cleanup and creates an enum for FHIR versions so that it isn't passed into every method. Also cleaned up BFD health check.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        return null;
+    }
+
+    public FhirContext getContext() {
+        return this.context;
+    }
+
+    public static FhirVersion fromUrl(String url) {
+        FhirVersion[] versions = FhirVersion.values();
+        for (FhirVersion v : versions) {
```
Comment (nflearl) [inline fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java:62]: nit: please no 1 letter variable names.

Alternative implementation (still with temporary variable):
return Stream.of(FhirVersion.values().anyMatch(fhir => url.contains(v.versionString));

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        if (this.jsonParser == null) {
+            EncodingEnum respType = EncodingEnum.forContentType(EncodingEnum.JSON_PLAIN_STRING);
+            this.jsonParser = respType.newParser(this.context);
+        }
+        return jsonParser;
+    }
+
+    public static FhirVersion from(FhirVersionEnum fhirVersionEnum) {
+        FhirVersion[] versions = FhirVersion.values();
+        for (FhirVersion v : versions) {
```
Comment (nflearl) [inline fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java:48]: nit: please no 1 letter variable names.

Alternative implementation (still with temporary variable):
return Stream.of(FhirVersion.values().anyMatch(fhir => fhir.fhirVersionEnum == fhirVersionEnum));

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            org.hl7.fhir.dstu3.model.ResourceType.Patient),
+    R4("org.hl7.fhir.r4.model", FhirContext.forR4(), FhirVersionEnum.R4, "/v2/",
+            org.hl7.fhir.r4.model.ResourceType.Patient);
+
+    private String classLocation;
+    private FhirContext context;
+    private IParser jsonParser;
+    private FhirVersionEnum fhirVersionEnum;
+    private String versionString;
+    private Object patientEnum;
```
Comment (nflearl) [inline fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java:27]: make final

**Review by nflearl — CHANGES_REQUESTED** [review]
Minor cleanups requested.

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java (hunk 3 above)
Comment (sharshar-sb) [reply fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java]: done

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java (hunk 2 above)
Comment (sharshar-sb) [reply fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java]: done

File: fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java (hunk 1 above)
Comment (sharshar-sb) [reply fhir/src/main/java/gov/cms/ab2d/fhir/FhirVersion.java]: done

**Review by sb-wnyffenegger — APPROVED** [review]

**Review by nflearl — APPROVED** [review]

**Pull request merged**